### PR TITLE
Added redmine

### PR DESCRIPTION
**Pull request by joshuacox**
Adding the project manager Redmine to the rancher templates

### CODE REVIEW

File: templates/redmine/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - /home/redmine/data
+      - /var/lib/mysql
+      - /var/log/redmine
+    labels:
+      io.rancher.container.start_once: true
+    entrypoint: ["/bin/true"]
+
+  db:
+    restart: always
+    image: mariadb
```
Comment (rawmind0) [inline templates/redmine/0/docker-compose.yml]: Please, use a concrete version instead implicit latest...

**Comment by rawmind0** [comment]
Hi @joshuacox .. Thanks so much for the PR....

Could you please take a look at change requests?? Thanks...

**Comment by joshuacox** [comment]
@rawmind0 sorry my notifications were not going through.

File: templates/redmine/0/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    environment:
+    - MYSQL_USER=redmine
+    - MYSQL_PASSWORD=${DB_PASS}
+    - MYSQL_ROOT_PASSWORD=${DB_PASS}
+    - MYSQL_DATABASE=redmine_production
+
+  redmine:
+    image: sameersbn/redmine:3.3.2-1
+    labels:
+      io.rancher.sidekicks: db,redmine-datavolume
```
Comment (rawmind0) [inline templates/redmine/0/docker-compose.yml]: db shouldn't be a sidekick.

File: templates/redmine/0/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    image: "busybox"
+    volumes:
+      - /home/redmine/data
+      - /var/lib/mysql
+      - /var/log/redmine
+    labels:
+      io.rancher.container.start_once: true
+    entrypoint: ["/bin/true"]
+
+  db:
```
Comment (rawmind0) [inline templates/redmine/0/docker-compose.yml]: db shouldn't be a sidekick. Also, if you use "volumes_from", db would need another busybox just for him.

File: templates/redmine/0/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    restart: always
+    image: mariadb:10.2
+    volumes_from:
+    - redmine-datavolume
```
Comment (rawmind0) [inline templates/redmine/0/docker-compose.yml]:
```
--    - redmine-datavolume
++    - db-datavolume
```

File: templates/redmine/0/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+version: '2'
+
+services:
+  redmine-datavolume:
+    image: "busybox"
+    volumes:
+      - /home/redmine/data
+      - /var/lib/mysql
```
Comment (rawmind0) [inline templates/redmine/0/docker-compose.yml]: Don't share volumes to more than one service.
```
--      - /var/lib/mysql
```

File: templates/redmine/0/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      - /var/log/redmine
+    labels:
+      io.rancher.container.start_once: true
+    entrypoint: ["/bin/true"]
+
```
Comment (rawmind0) [inline templates/redmine/0/docker-compose.yml]: Add a docker volume just for db. Different services decouple storage.
```
+  db-datavolume:
+    image: "busybox"
+    volumes:
+      - /var/lib/mysql
+    labels:
+      io.rancher.container.start_once: true
+    entrypoint: ["/bin/true"]
```

File: templates/redmine/0/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    - DB_HOST=db
+    - DB_USER=redmine
+    - DB_NAME=redmine_production
+    - REDMINE_SECRET_TOKEN=${REDMINE_SECRET_TOKEN}
+    - SMTP_ENABLED=false
+    - IMAP_ENABLED=false
+    volumes_from:
+    - redmine-datavolume
+    ports:
+    - ${PORT}:80
```
Comment (rawmind0) [inline templates/redmine/0/docker-compose.yml]: Better expose port using lb.
```
--    ports:
--    - ${PORT}:80

++  redmine-lb:
++    image: rancher/lb-service-haproxy:v0.6.4
++    ports:
++      - ${PORT}:${PORT}
```

File: templates/redmine/0/rancher-compose.yml [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    description: "REDMINE_SECRET_TOKEN"
+    label: "REDMINE_SECRET_TOKEN"
+    required: true
+    default: "REALLYlongrandomhashkjekkckksixci"
+    type: "string"
+
+services:
+  redmine:
+    scale: 1
+    retain_ip: true
```
Comment (rawmind0) [inline templates/redmine/0/rancher-compose.yml]: Add healthcheck and lb section
```
++    health_check:
++      healthy_threshold: 2
++      response_timeout: 5000
++      port: 80
++      unhealthy_threshold: 3
++      interval: 5000
++      strategy: recreate
++      request_line: GET / # HTTP/1.0
++  redmine-lb:
++    scale: 1
++    start_on_create: true
++    lb_config:
++      certs: []
++      port_rules:
++      - priority: 1
++        protocol: http
++        service: redmine
++        source_port: ${PORT}
++        target_port: 80
```

**Comment by rawmind0** [comment]
Hi @joshuacox ,

sorry i missed the db as a sidekick in the previous review. Please, take a look at the change request.

File: templates/redmine/0/docker-compose.yml (hunk 2 above)
Comment (joshuacox) [inline templates/redmine/0/docker-compose.yml]: done

File: templates/redmine/0/docker-compose.yml (hunk 5 above)
Comment (joshuacox) [inline templates/redmine/0/docker-compose.yml]: on all of this volume stuff, how about I ditch the busybox container and use the version 2 volumes: block?  Pushing an example next

File: templates/redmine/0/docker-compose.yml (hunk 7 above)
Comment (joshuacox) [inline templates/redmine/0/docker-compose.yml]: how about I just prep it for traefik?

File: templates/redmine/0/rancher-compose.yml (hunk 8 above)
Comment (joshuacox) [inline templates/redmine/0/rancher-compose.yml]: if I set it up for traefik do I still want to throw an LB in between traefik and redmine?

Comment (rawmind0) [inline templates/redmine/0/rancher-compose.yml]: If you use traefik, it's just fine. The point is avoid the expose of ports directly from the service. Could you please add a README.md explaining that the package needs a traefik just for accessing??

**Comment by rawmind0** [comment]
Thanks @joshuacox ....

LGTM....

**Pull request merged**